### ISSUE TITLE
Pinniped operator can easily learn Pinniped version from running Pinniped instance

### ISSUE THREAD

**Issue by ankeesler** [vmware-tanzu/pinniped#717]
**Is your feature request related to a problem? Please describe.**

- As a user of a product that installs Pinniped on my behalf (e.g., TMC, TKG), I would like to be able to understand what version of Pinniped (all components: concierge + supervisor + CLI) is running in my cluster, so that I can confirm whether or not I am running a version that has vulnerabilities/known bugs.

**Describe the solution you'd like**

- Ideally, I'd love to be able to access this version information via an HTTP call.
- Also ideally, I'd love to be able to access this version information in the logs.

**Describe alternatives you've considered**

- I looked at the logs and this is what I saw.
```
I0715 15:10:15.603680       1 main.go:366] Running pinniped-supervisor/v0.0.0 (linux/amd64) kubernetes/$Format at version.Info{Major:"", Minor:"", GitVersion:"v0.0.0-master+$Format:%h$", GitCommit:"$Format:%H$", GitTreeState:"", BuildDate:"2021-06-14T20:08:22Z", GoVersion:"go1.15.12", Compiler:"gc", Platform:"linux/amd64"}
```

**Are you considering submitting a PR for this feature?**

Mmmm, maybe? Not right this instance, but if y'all need help let me know.

- **How will this project improvement be tested?**
- Integration tests?
- **How does this change the current architecture?**
- Hopefully not...
- **How will this change be backwards compatible?**
- Net new functionality...
- **How will this feature be documented?**
- On pinniped.dev? Maybe in one of the existing docs?

**Additional context**

**Comment by enj** [comment]
The (incorrect) concierge information is already available via the service `/version` endpoint.  We just need to fill that information out when we build the binaries.

```shell
kubectl port-forward service/pinniped-concierge-api -n concierge 8443:443
```

```shell
$ curl -k https://localhost:8443/version
```

```json
{
  "major": "",
  "minor": "",
  "gitVersion": "v0.0.0-master+$Format:%H$",
  "gitCommit": "$Format:%H$",
  "gitTreeState": "",
  "buildDate": "2021-06-11T18:58:01Z",
  "goVersion": "go1.16.5",
  "compiler": "gc",
  "platform": "linux/amd64"
}
```

**Comment by ankeesler** [comment]
Note: I think right now, [when `hack/get-ldflags.sh` runs during the container build](https://github.com/vmware-tanzu/pinniped/blob/d204b46c185bf68155e45ff575331391fe1602ed/Dockerfile#L24), we cannot get access to git information since the `.git` directory is ignored in the [`COPY . .`](https://github.com/vmware-tanzu/pinniped/blob/d204b46c185bf68155e45ff575331391fe1602ed/Dockerfile#L9) build line.

**Comment by cfryanr** [comment]
A first cut at this was implemented in #724. It adds the git commit SHA, and with a little homework a user can figure out which version is running based on the git commit SHA.

**Comment by joshuatcasey** [comment]
This build information is now available.
